packages/ra-ui-materialui/src/button/SaveButton.spec.tsx

```diff
@@ -337,4 +337,52 @@ describe('<SaveButton />', () => {
             });
         });
     });
+
+    it('should disable <SaveButton/> if an input is being validated asynchronously', async () => {
+        const dataProvider = ({
+            getOne: () =>
+                Promise.resolve({
+                    data: { id: 123, title: 'lorem' },
+                }),
+            update: (_, { data }) => Promise.resolve({ data }),
+        } as unknown) as DataProvider;
+
+        const validateAgainstServer = value =>
+            new Promise(resolve => setTimeout(() => value, 400));
+
+        const validateAsync = async (value, allValues) => {
```

Comment:
I don't understand why you need 2 separate functions (validateAgainstServer and validateAsync). It make sthe code harder to read.